src/Components/PropertyInspector/PropertyInspectoryModel.ts

```diff
@@ -344,21 +384,116 @@ abstract class PropertyInspectorModel {
                                                   .mapValue,
                                               path: PropertyInspectorModel.buildPath(
                                                   path,
-                                                  modelProperty.name
+                                                  modelProperty.name,
+                                                  modelProperty.index
                                               ),
                                               propertySourceObject: mapValue,
                                               mapInfo: { key },
                                               isMapChild: true,
                                               forceSet: true,
-                                              schemas
+                                              schemas,
+                                              isArrayItem
                                           }
                                       );
                                   })
                                 : null,
                         ...(schemas && { mapSchemas: schemas })
                     };
                 }
-                case DTDLSchemaType.Array: // TODO support arrays in future
+                case DTDLSchemaType.Array: {
+                    const arrayItems: any[] = PropertyInspectorModel.getPropertyValueOrDefault(
+                        modelProperty.name,
+                        propertySourceObject,
+                        dtdlPropertyTypesEnum.Array
+                    );
+                    const targetSchema = schemas?.find(
+                        (s) =>
+                            s['@id'] ===
+                            ((modelProperty.schema as any).elementSchema ||
+                                modelProperty.schema)
+                    );
+                    return {
+                        name: modelProperty.name,
+                        displayName: PropertyInspectorModel.parsePropertyTreeDisplayName(
+                            {
+                                name: modelProperty.name,
```

Comment:
why do this spread here? If we pull name off the object and then spread the object after that, do we gain anything? Why not just pass the object itself?

Reply:
I was having issues with the transpiler not understanding that `name` is a property on `modelProperty`. It seems more happy with passing it directly now. Will remove the extra ops.